### PR TITLE
Fixes some glitches with auto-rotation of vis_overlays

### LINKED ISSUE #42138: Airtight plastic flaps occasionally unload their sprite when unscrewed from floor
Issue reported from  Round ID: 99460 (/tg/Station Sybil [ENGLISH] [US-EAST] [100% LAG FREE])

Reporting client version: 512

Problem Description: The sprite for plastic flaps will become invisible after the flaps are unscrewed from the floor. They are still able to be clicked on, interacted with, and pulled, but have an invisible sprite. 

What did you expect to happen: The plastic flaps to be visible upon their unscrewing from the floor.

What happened instead: The sprite seemingly unloads, and almost acts as an invisible vision barrier. It can be moved, and interacted with as normal.

Why is this bad/What are the consequences: While not an exceedingly large issue, it could have some odd implications in the sense of creating invisible barriers to players where otherwise unexpected, and be completely unrelated to mimery. 

Steps to reproduce the problem: Simply unscrew the airtight plastic flaps from the floor. However, you may or may not have to pull the flaps from their original position as well.

Server: Sybil

Revision: 
Server-Side Code Modifications: https://tgstation13.org/parsed-logs/basil/server_side_modifications.dm

Round ID: 99460
BYOND Version: 512.1453
Server revision compiled on: 
Master commit: 66c14d45513138a808e607170c6bac74e3c7a42f
The following pull requests are currently test merged:
#40907: 'Readds Birdboatstation (ORANGES SAYS DO NOT TEST MERGE)' by Bawhoppen at commit 348e864820
#41994: 'UNSTOPPABLE flag is now checked every time something tries to pass something instead of just once in the Move() proc' by kevinz000 at commit 866deec6a3

Local commit: 027367ab316c2af84359492daba71e5c01192a5d
Server tools version: /tg/station 13 Server v3.2.6.2

Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?): There was nothing especially abnormal with my character, standard assistant gear: Insulated gloves, sunglasses, welding helmet, cigar. Nothing out of the ordinary aside from an admin spawning a crab earlier in the round. The round type was traitor + blood brothers.

round end panel

dreamseeker_2018-12-30_03-54-16

dreamseeker_2018-12-30_03-53-52

dreamseeker_2018-12-30_03-52-22

### PR DESCRIPTION
**Pull request by Menshin**
## About The Pull Request

Fixes some glitches and oversights with `vis_overlays` that prevented proper overlays auto-rotation.
Fixes #42138.

## Changelog
:cl: Menshin
fix: Plastic flaps won't become invisible on unscrew anymore
/:cl:

### CODE REVIEW

**Comment by fizzle7** [comment]
Relevant runtime:

```
[2019-02-01 04:49:35.095] runtime error: Cannot read null.icon
 - proc name: rotate vis overlay (/datum/controller/subsystem/vis_overlays/proc/rotate_vis_overlay)
 -   source file: vis_overlays.dm,74
 -   usr: 0
 -   src: Vis contents overlays (/datum/controller/subsystem/vis_overlays)
 -   call stack:
 - Vis contents overlays (/datum/controller/subsystem/vis_overlays): rotate vis overlay(the airtight plastic flaps (/obj/structure/plasticflaps), 1, 1)
 - /datum/callback (/datum/callback): InvokeAsync(the airtight plastic flaps (/obj/structure/plasticflaps), 1, 1)
 - the airtight plastic flaps (/obj/structure/plasticflaps):  SendSignal("atom_dir_change", /list (/list))
```

Caused by flaps' managed_vis_overlay getting deleted after moving it. See if your code fixes it. Unscrew the flaps, move them and then wait a few minutes to see if GC deleted the overlay.

**Comment by Menshin** [comment]
I've let `vis_overlay` generates for the four directions, then `qdel'ed` after 2 minutes (except the direction still used, of course), twice.
No runtime.

Double checking in VV : the flaps `managed_vis_overlay` and the global `vis_overlay_cache` are properly updated.

File: code/controllers/subsystem/vis_overlays.dm [hunk 1]
```diff
@@ -67,10 +67,14 @@ SUBSYSTEM_DEF(vis_overlays)
 		UnregisterSignal(thing, COMSIG_ATOM_DIR_CHANGE)
 
 /datum/controller/subsystem/vis_overlays/proc/rotate_vis_overlay(atom/thing, old_dir, new_dir)
+	if(old_dir == new_dir)
+		return
 	var/rotation = dir2angle(old_dir) - dir2angle(new_dir)
 	var/list/overlays_to_remove = list()
-	for(var/i in thing.managed_vis_overlays)
+	//the next loop is going to mess with thing.managed_vis_overlays, so iterate on a copy of it
```
Comment (SpaceManiac) [inline code/controllers/subsystem/vis_overlays.dm]: This comment is inaccurate, this local variable isn't actually a copy

Comment (Menshin) [reply code/controllers/subsystem/vis_overlays.dm]: You're right, it's superfluous, the `for in` loop already makes and interates on a copy of `things.managed_vis_overlays`
Thanks.

**Review by SpaceManiac — APPROVED** [review]
Does in fact fix the bug

**Pull request merged**